### PR TITLE
Feature/xss

### PR DESCRIPTION
**Pull request by afeena**
Basic xss, inject xss into page, get data from GET and POST requests

### CODE REVIEW

File: server.py [hunk 1]
```diff
@@ -44,28 +48,28 @@ def _make_response(self, msg):
...
+            # dummy for wp-content
+            if re.match(r'/wp-content/.*', path):
+                m = self._make_response(msg=dict(detection={'name': 'wp-content', 'order': 1}))
+                return m
+            sensor_uuid = data['uuid'] if 'uuid' in data else None
+        except (TypeError, ValueError, KeyError) as e:
+            print('error parsing: {}'.format(data))
+            m = self._make_response(msg=type(e).__name__)
+        else:
+            session = yield from HttpRequestHandler.session_manager.add_or_update_session(data)
```
Comment (glaslos) [inline server.py]: Shouldn't that be before the wp-content match?

File: xss_emulator.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import mimetypes
+
+
+class XSSemulator:
+    def extract_xss_data(self, data):
+        xss = None
+        if 'post_data' in data:
+            for field, val in data['post_data'].items():
+                val = urllib.parse.unquote(val)
+                xss = re.match(r'.*<(.*)>.*', val)
```
Comment (glaslos) [inline xss_emulator.py]: I think we should centralize all patterns. Can you create an issue for that?

Comment (afeena) [inline xss_emulator.py]: Sure :)

File: server.py (hunk 1 above)
Comment (afeena) [inline server.py]: Do we need store wp-content request as a separate session?

Comment (glaslos) [inline server.py]: I think we should store it as session. Right now you return before you store the session. You think we shouldn't store it as session?

Comment (afeena) [inline server.py]: I thought a bit about it and you are right. fixed

**Pull request merged**